### PR TITLE
Prevent implicit down cast. This may give compiler warnings.

### PR DESCRIPTION
**Pull request by jokri**
This is a fix for issue #1133.

### CODE REVIEW

**Comment by jokri** [comment]
I added my alternate/commit E-mail address to my google account.

File: googlemock/test/gmock-generated-matchers_test.cc [hunk 1]
```diff
@@ -120,21 +120,21 @@ TEST(ArgsTest, AcceptsOneTemplateArg) {
 }
 
 TEST(ArgsTest, AcceptsTwoTemplateArgs) {
-  const tuple<short, int, long> t(4, 5, 6L);  // NOLINT
+  const tuple<short, int, long> t(static_cast<short>(4), 5, 6L);  // NOLINT
```
Comment (pavelkryukov) [inline googlemock/test/gmock-generated-matchers_test.cc]: Is "NOLINT" still required?

Comment (jokri) [inline googlemock/test/gmock-generated-matchers_test.cc]: Maybe not. How can I tell? Is lint run during the "check" process?

**Comment by gennadiycivil** [comment]
Could you please provide clear test evidence before and after

**Comment by jokri** [comment]
Is this the test evidence you are looking/asking for?

`...
<Project DefaultTargets="Build" ToolsVersion="15.0" xmlns="http://schemas.microsoft.com/developer/msbuild/2003">
...
    <PlatformToolset>v141</PlatformToolset>
...
      <WarningLevel>Level4</WarningLevel>
      <TreatWarningAsError>true</TreatWarningAsError>
...


before:
1>------ Rebuild All started: Project: gmock, Configuration: Debug Win32 ------
1>gmock-all.cc
1>gtest-all.cc
1>gmock.vcxproj -> H:\contribute\googletest-jokri\googlemock\msvc\2017\Debug\gmock.lib
2>------ Rebuild All started: Project: gmock_main, Configuration: Debug Win32 ------
2>gmock_main.cc
2>gmock_main.vcxproj -> H:\contribute\googletest-jokri\googlemock\msvc\2017\Debug\gmock_main.lib
3>------ Rebuild All started: Project: gmock_test, Configuration: Debug Win32 ------
3>gmock_all_test.cc
3>C:\Program Files (x86)\Microsoft Visual Studio\2017\Community\VC\Tools\MSVC\14.10.25017\include\tuple(165): error C2220: warning treated as error - no 'object' file generated
3>C:\Program Files (x86)\Microsoft Visual Studio\2017\Community\VC\Tools\MSVC\14.10.25017\include\tuple(292): note: see reference to function template instantiation 'std::_Tuple_val<_This>::_Tuple_val<_Ty>(_Other &&)' being compiled
3>        with
3>        [
3>            _This=short,
3>            _Ty=int,
3>            _Other=int
3>        ]
3>C:\Program Files (x86)\Microsoft Visual Studio\2017\Community\VC\Tools\MSVC\14.10.25017\include\tuple(291): note: see reference to function template instantiation 'std::_Tuple_val<_This>::_Tuple_val<_Ty>(_Other &&)' being compiled
3>        with
3>        [
3>            _This=short,
3>            _Ty=int,
3>            _Other=int
3>        ]
3>C:\Program Files (x86)\Microsoft Visual Studio\2017\Community\VC\Tools\MSVC\14.10.25017\include\tuple(422): note: see reference to function template instantiation 'std::tuple<short,int,long>::tuple<std::_Exact_args_t,_Ty,_Ty,long,0>(_Tag,_This2 &&,_Ty &&,long &&)' being compiled
3>        with
3>        [
3>            _Ty=int,
3>            _Tag=std::_Exact_args_t,
3>            _This2=int
3>        ]
3>C:\Program Files (x86)\Microsoft Visual Studio\2017\Community\VC\Tools\MSVC\14.10.25017\include\tuple(421): note: see reference to function template instantiation 'std::tuple<short,int,long>::tuple<std::_Exact_args_t,_Ty,_Ty,long,0>(_Tag,_This2 &&,_Ty &&,long &&)' being compiled
3>        with
3>        [
3>            _Ty=int,
3>            _Tag=std::_Exact_args_t,
3>            _This2=int
3>        ]
3>H:\contribute\googletest-jokri\googlemock\test/gmock-generated-matchers_test.cc(123): note: see reference to function template instantiation 'std::tuple<short,int,long>::tuple<int,int,long,0>(_This2 &&,int &&,long &&)' being compiled
3>        with
3>        [
3>            _This2=int
3>        ]
3>H:\contribute\googletest-jokri\googlemock\test/gmock-generated-matchers_test.cc(123): note: see reference to function template instantiation 'std::tuple<short,int,long>::tuple<int,int,long,0>(_This2 &&,int &&,long &&)' being compiled
3>        with
3>        [
3>            _This2=int
3>        ]
3>C:\Program Files (x86)\Microsoft Visual Studio\2017\Community\VC\Tools\MSVC\14.10.25017\include\tuple(165): warning C4244: 'initializing': conversion from 'int' to 'short', possible loss of data
3>Done building project "gmock_test.vcxproj" -- FAILED.
========== Rebuild All: 2 succeeded, 1 failed, 0 skipped ==========


after:

1>------ Rebuild All started: Project: gmock, Configuration: Debug Win32 ------
1>gmock-all.cc
1>gtest-all.cc
1>gmock.vcxproj -> H:\contribute\googletest-jokri\googlemock\msvc\2017\Debug\gmock.lib
2>------ Rebuild All started: Project: gmock_main, Configuration: Debug Win32 ------
2>gmock_main.cc
2>gmock_main.vcxproj -> H:\contribute\googletest-jokri\googlemock\msvc\2017\Debug\gmock_main.lib
3>------ Rebuild All started: Project: gmock_test, Configuration: Debug Win32 ------
3>gmock_all_test.cc
3>gmock_test.vcxproj -> H:\contribute\googletest-jokri\googlemock\msvc\2017\Debug\gmock_test.exe
3>gmock_test.vcxproj -> H:\contribute\googletest-jokri\googlemock\msvc\2017\Debug\gmock_test.pdb (Full PDB)
========== Rebuild All: 3 succeeded, 0 failed, 0 skipped ==========
`

**Comment by KindDragon** [comment]
Can be closed. PR #1218 fixed this

**Comment by jokri** [comment]
Closing this PR, PR #1218 fixed this.

**Pull request closed without merging**